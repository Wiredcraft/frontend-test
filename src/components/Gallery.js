import React, { Component } from "react";
import { connect } from "react-redux";

import { waitForImages } from "../utils/responsiveMasonry";

class Gallery extends Component {
    componentDidMount() {
        waitForImages();
    }
    componentDidUpdate() {
        waitForImages();
    }
    render() {
        const { imgList } = this.props;
        return (
            <div className="grid-container">
                {imgList.map((item, index) => {
                    if (item) {
                        return (
                            <div className="grid-item" key={item._id}>
                                <img
                                    className="grid-content"
                                    src={item.src}
                                    alt={item.src}
                                ></img>
                            </div>
                        );
                    }
                    return null;
                })}
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    imgList: state.gallery.filteredImgList
});

export default connect(mapStateToProps)(Gallery);
