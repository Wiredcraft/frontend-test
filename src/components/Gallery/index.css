.wc-columns {
    column-count: 6;
    column-gap: 1.25rem;
    orphans: 1;
    widows: 1;
}

.wc-columns .card {
    margin-bottom: .75rem;
    border-radius: 0.75rem;
    border: none;
}

.wc-columns .card .wc-picture{
    padding: 0;
    border: none;
}