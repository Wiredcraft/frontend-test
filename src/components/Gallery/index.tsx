// eslint-disable-next-line
import React, { useState, useEffect } from 'react';
import './index.css';
import {getPictures} from './service';

export type LoadingProps = {
  tip?: string;
};

export type pictureCard = {
  id: string;
  index: number;
  name: string;
  src: string;
};

const Gallery: React.FC<LoadingProps> = (props) => {
  const [isLoading, setIsLoading] = useState(false);
  const [pictureList, setPictureList] = useState<pictureCard[]>([]);

  useEffect(() => {
    setIsLoading(true);
    const fetchData = async () => {
      const pictures = await getPictures();
      console.log(pictures.data);
      setPictureList(pictures.data);
      setIsLoading(false);
    };
    fetchData();
  }, []);
  return (
    <div className="container">
      {isLoading ? 'loading' : ''}
      <div className="wc-columns">
        {pictureList.map((item) => (
          <div key={item.id} className="card">
           <img src={item.src}  className="img-thumbnail wc-picture" />
          </div>
        ))}
      </div>
    </div>
  );
};

export default Gallery;
