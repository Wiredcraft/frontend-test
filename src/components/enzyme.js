import 'regenerator-runtime/runtime';
import Enzyme, { configure, shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
export { shallow, mount, render };
export default Enzyme;
