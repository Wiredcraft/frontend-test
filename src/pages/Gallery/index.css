.wc-columns {
    column-count: 6;
    column-gap: 1.25rem;
    orphans: 1;
    widows: 1;
}

.wc-columns .card {
    width: 100%;
    margin-bottom: 0.75rem;
    border-radius: 1rem;
    border: none;
    -webkit-column-break-inside: avoid;
    break-before: avoid;
}

.wc-columns .card .wc-picture {
    position: absolute;
    top: 0;
    left: 0;
    padding: 0;
    border: none;
    border-radius: 0.75rem;
}

.wc-image-placeholder {
    width: 100%;
    border-radius: 0.75rem;
}