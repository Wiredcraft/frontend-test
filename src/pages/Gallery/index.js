import React, { useEffect, useState } from 'react';
import Card from '@/components/Card';
import { getImages } from '@/utils/axios/api.js';
import './index.scss';

export default function Gallery() {
  const [images, setImages] = useState([]);
  useEffect(async () => {
    const res = await getImages();
    setImages(res.data);
  }, []);

  const displayCards = images.map((img) => (
    <Card key={img._id} src={img.src} alt={img.name} />
  ));
  return <div className="gallery">{displayCards}</div>;
}
