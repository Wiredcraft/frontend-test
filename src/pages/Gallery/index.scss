.gallery {
  max-width: 1200px;
  padding: 18px 20px;
  margin: 0 auto;
  columns: 6;
  column-gap: 14px;

  @media screen and (max-width: 576px) {
    columns: 2;
  }

  @media screen and (min-width: 576px) and (max-width: 768px) {
    columns: 3;
  }

  @media screen and (min-width: 768px) and (max-width: 992px) {
    columns: 4;
  }
}
