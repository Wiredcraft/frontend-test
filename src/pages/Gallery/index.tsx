// eslint-disable-next-line
import React, { useState, useEffect } from 'react';
import './index.css';
import Card from '../../components/Card';
import SearchBar from '../../components/SearchBar';
import { getPictures } from './service';
import { pictureCard } from '../../data.d';

const Gallery = () => {
  const [isLoading, setIsLoading] = useState(false);
  const [pictureList, setPictureList] = useState<pictureCard[]>([]);
  const [filteredPictureList, setFilteredPictureList] = useState<pictureCard[]>([]);
  const [searchQuery, setSearchQuery] = useState('');

  const updateSearch = async (searchQuery: string) => {
    setFilteredPictureList(pictureList);
    setSearchQuery(searchQuery);
  }

  const onSearchButtonClick = () => {
    const filtered = pictureList.filter((item: pictureCard) => {
      return item.name === searchQuery;
    })
    setFilteredPictureList(filtered);
  }
 
  useEffect(() => {
    setIsLoading(true);
    const fetchData = async () => {
      const pictures = await getPictures();
      setPictureList(pictures.data);
      setFilteredPictureList(pictures.data);
      setIsLoading(false);
    };
    fetchData();
  }, []);

  return (
    <div>
      <div className="container mt-5">
        <div className="row justify-content-end">
          <div className="col-auto">
          <SearchBar
            searchQuery={searchQuery}
            setSearchQuery={updateSearch}
          />
          </div>
          <div className="col-auto">
            <button 
              className="btn btn-outline-secondary" 
              type="button" 
              id="button-addon2"
              onClick={onSearchButtonClick} 
            >Search</button>
          </div>
        </div>
        {isLoading ? 'loading' : ''}
        <div className="wc-columns">
          {filteredPictureList.map((item) => (
            <Card src={item.src} id={item.id} />
          ))}
        </div>
      </div>
    </div>
  );
};

export default Gallery;
