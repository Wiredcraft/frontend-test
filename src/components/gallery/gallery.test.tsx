/**
 * Gallery component test
 * Fri Jul 24 16:36:15 2020
 * by xiaoT
 */
import React, { FC } from 'react'
import ReactDom from 'react-dom'
import { shallow, mount } from 'enzyme'

import mountWithProvider from '@Store/mockStore'
import Gallery from './gallery'

const images = [
  {
    _id: '5f115174961c75468fbe0f44',
    index: 0,
    name: 'purple',
    src: 'https://picsum.photos/240/379?random=371',
  },
  {
    _id: '5f115174ad8e8ef73a0dab9e',
    index: 1,
    name: 'purple',
    src: 'https://picsum.photos/240/392?random=286',
  }
]

const defaultGridRowRect = {
  gridRowGap: 10,
  gridRowHeight: 350,
  gridColumnWidth: 245
}

describe('Gallery Test', () => {
  let galleryComponent: any

  beforeEach(() => {
    galleryComponent  = mountWithProvider(<Gallery gridRowRect={defaultGridRowRect} images={images} />)()
  })
  afterEach(() => {
    galleryComponent.unmount()
  })
  // should be match the snapshot
  it('should be match the snapshot', () => {
    expect(galleryComponent.html()).toMatchSnapshot()
  })
  // Gallery should be have a NavBar compoent
  it('should be have a NavBar', () => {
    const NavBar = galleryComponent.find('NavBar')

    expect(NavBar.length).toEqual(1)
  })
  // Gallery should be have two Image compoent
  it('Gallery should be have two Image compoent', () => {
    const Image = galleryComponent.find('Image')

    expect(Image.length).toEqual(images.length)
  })
  // Gallery prop gridRowRect shoule be equql defaultGridRowRect
  it('Gallery prop gridRowRect shoule be equql defaultGridRowRect', () => {
    const gridRowRectProp = galleryComponent.props().children.props.gridRowRect

    expect(gridRowRectProp).toEqual(defaultGridRowRect)
  })
})