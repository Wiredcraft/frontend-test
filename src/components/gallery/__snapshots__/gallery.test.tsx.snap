// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery Test should be match the snapshot 1`] = `"<div class=\\"nav-bar\\"><div data-justify=\\"flex-end\\" data-align=\\"center\\" class=\\"nav-bar-inner flex\\"><form><label data-align=\\"center\\" class=\\"form-item flex\\"><i class=\\"icon icon-search\\"></i><input type=\\"text\\" value=\\"\\" name=\\"searchKey\\" class=\\"flex-1\\" placeholder=\\"请输入关键字\\"/></label></form><i class=\\"icon icon-home\\"></i><i class=\\"icon icon-bell\\"></i><i class=\\"icon icon-avatar\\"></i></div></div><div class=\\"gallery\\"><div style=\\"grid-row-end:span NaN\\" class=\\"image-container\\"><img class=\\"image\\" data-lazy=\\"https://gblobscdn.gitbook.com/spaces%2F-LceGMK-Zxa6_-QeGdy1%2Favatar.png?alt=media\\" src=\\"https://picsum.photos/240/379?random=371\\" alt=\\"\\"/></div><div style=\\"grid-row-end:span NaN\\" class=\\"image-container\\"><img class=\\"image\\" data-lazy=\\"https://gblobscdn.gitbook.com/spaces%2F-LceGMK-Zxa6_-QeGdy1%2Favatar.png?alt=media\\" src=\\"https://picsum.photos/240/392?random=286\\" alt=\\"\\"/></div></div>"`;
